app/models/user.rb

```diff
@@ -1,5 +1,5 @@
 class User < ActiveRecord::Base
-  has_many :milestones, through: :milestone_users
+  has_many :milestones, -> { order :referals_count}, through: :milestone_users
   has_many :milestone_users, dependent: :delete_all
   has_many :referals, class_name: User, foreign_key: :inviter_id
   belongs_to :inviter, class_name: User
```

Comment:
all_milestones method:
returns array of hashes
{milestone: Milestone.new, reached: false, milestone_user: nil}